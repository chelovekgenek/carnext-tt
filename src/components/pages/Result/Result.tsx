import React from "react"
import { connect } from "react-redux"

import { getPromoted } from "store/entities/promotion"
import { Layout } from "components/App"
import { TAppState } from "store/entities"

interface IStateProps {
  promoted: ReturnType<typeof getPromoted>
}
interface IProps extends IStateProps {}

export const Result: React.FC<IProps> = ({ promoted }) => (
  <Layout>
    {promoted ? (
      <React.Fragment>
        <h1>Congratulations, you won!</h1>
        <p>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore
          magna aliqua.
        </p>
      </React.Fragment>
    ) : (
      <React.Fragment>
        <h1>Better luck next time!</h1>
        <p>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore
          magna aliqua.
        </p>
      </React.Fragment>
    )}
  </Layout>
)

export default connect<IStateProps, {}, {}, TAppState>(state => ({
  promoted: getPromoted(state),
}))(Result)
