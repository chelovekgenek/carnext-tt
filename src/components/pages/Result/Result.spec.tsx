import React from "react"
import { shallow } from "enzyme"

import { Result } from "./Result"

describe("Result", () => {
  it("should match snapshot if promoted is true", () => {
    expect(shallow(<Result promoted={true} />)).toMatchSnapshot()
  })
  it("should match snapshot if promoted is false", () => {
    expect(shallow(<Result promoted={false} />)).toMatchSnapshot()
  })
})
