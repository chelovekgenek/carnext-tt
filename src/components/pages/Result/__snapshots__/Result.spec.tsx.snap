// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Result should match snapshot if promoted is false 1`] = `
<Layout>
  <h1>
    Better luck next time!
  </h1>
  <p>
    Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
  </p>
</Layout>
`;

exports[`Result should match snapshot if promoted is true 1`] = `
<Layout>
  <h1>
    Congratulations, you won!
  </h1>
  <p>
    Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
  </p>
</Layout>
`;
